anchore_engine/services/policy_engine/engine/policy/gates/vulnerabilities.py

```diff
@@ -309,9 +330,9 @@ def evaluate(self, image_obj, context):
 
                     parameter_data['severity'] = pkg_vuln.vulnerability.severity.upper()
                     parameter_data['vulnerability_id'] = pkg_vuln.vulnerability_id
-                    parameter_data['pkg_class'] = 'os'
                     parameter_data['pkg_type'] = pkg_vuln.pkg_type
-                    
+                    parameter_data['pkg_class'] = determine_pkg_class(pkg_vuln)
```

Comment:
I thought the same and wasn't sold on how this is being solved here. All the current Github vulnerabilities are reporting their `self.package_type` as `'all'`, and never as `'non-os'`, and it does this within an `if` condition which only inspects `cpe_vulnerabilities` for `non-os` and NVD for `os` - an impossible condition to meet because all Github vulnerabilities use NVD. 

An alternative would be to use `self.package_type` when it is `non-os` or `os` and omit it when it says `'all'` ?